### PR TITLE
Add cases to cover luks vol-download and vol-upload

### PR DESCRIPTION
**Pull request by lento-sun**
Sign-offed by: Yi Sun (lentosun@163.com)

### CODE REVIEW

**Comment by lento-sun** [comment]
Test with libvirt-3.1.0-2.el7.x86_64
Failed cases are due to libvirt's behavior changed during 3.0.0-x, and we need to update 
avocado-vt/virttest/utils_test/libvirt.py:pre_pool(), which is being fixed. 

## avocado run --vt-type libvirt type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt..
JOB ID     : 3f8cd8a465d178e57dad0cee3ccc51127dd89947
JOB LOG    : /root/avocado/job-results/job-2017-03-28T14.18-3f8cd8a/job.log
TESTS      : 32
 (01/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.download.non_acl.0-1024: PASS (6.66 s)
 (02/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.download.non_acl.1024-4096: PASS (6.76 s)
 (03/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.download.non_acl.0-end: PASS (8.88 s)
 (04/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.download.non_acl.1024-end: PASS (6.72 s)
 (05/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.download.non_acl.no_options: PASS (9.17 s)
 (06/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.upload.non_acl.0-end: PASS (6.76 s)
 (07/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.upload.non_acl.1024-end: PASS (8.89 s)
 (08/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.upload.non_acl.no_options: PASS (6.82 s)
 (09/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.download.non_acl.0-1024: PASS (20.45 s)
 (10/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.download.non_acl.1024-4096: PASS (15.41 s)
 (11/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.download.non_acl.0-end: PASS (19.23 s)
 (12/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.download.non_acl.1024-end: PASS (14.53 s)
 (13/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.download.non_acl.no_options: PASS (20.24 s)
 (14/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.upload.non_acl.0-end: PASS (14.60 s)
 (15/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.upload.non_acl.1024-end: PASS (20.46 s)
 (16/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.fs_pool.upload.non_acl.no_options: PASS (14.64 s)
 (17/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.download.non_acl.0-1024: FAIL (15.53 s)
 (18/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.download.non_acl.1024-4096: FAIL (9.46 s)
 (19/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.download.non_acl.0-end: FAIL (15.45 s)
 (20/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.download.non_acl.1024-end: FAIL (9.46 s)
 (21/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.download.non_acl.no_options: FAIL (15.51 s)
 (22/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.upload.non_acl.0-end: FAIL (9.41 s)
 (23/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.upload.non_acl.1024-end: FAIL (14.55 s)
 (24/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.disk_pool.upload.non_acl.no_options: FAIL (9.46 s)
 (25/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.download.non_acl.0-1024: PASS (7.71 s)
 (26/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.download.non_acl.1024-4096: PASS (7.83 s)
 (27/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.download.non_acl.0-end: PASS (7.68 s)
 (28/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.download.non_acl.1024-end: PASS (7.68 s)
 (29/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.download.non_acl.no_options: PASS (7.67 s)
 (30/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.upload.non_acl.0-end: PASS (7.69 s)
 (31/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.upload.non_acl.1024-end: PASS (7.70 s)
 (32/32) type_specific.lento.virsh.vol_download_upload.download_upload.luks_encrypt.netfs_pool.upload.non_acl.no_options: PASS (7.67 s)
RESULTS    : PASS 24 | ERROR 0 | FAIL 8 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2017-03-28T14.18-3f8cd8a/html/results.html
TESTS TIME : 360.66 s

**Comment by lento-sun** [comment]
1. logical pool doesn't support to create encrypted volume
2. acl test is covering pool/volume in this set of cases, and secret needs another acl, which will cause conflict in current cfg file

So, 1 and 2 excluded in config file

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py [hunk 1]
```diff
@@ -54,6 +58,55 @@ def write_file(path):
...
+    sec_xml.usage = 'volume'
+    sec_xml.volume = vol_path
+    sec_xml.xmltreefile.write()
+
+    ret = virsh.secret_define(sec_xml.xml)
+    utlv.check_exit_status(ret)
+    try:
+        encryption_uuid = re.findall(r".+\S+(\ +\S+)\ +.+\S+",
+                                     ret.stdout)[0].lstrip()
+    except IndexError, e:
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: put e in line 80 error msg or remove

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py [hunk 2]
```diff
@@ -54,6 +58,55 @@ def write_file(path):
     write_fd.close()
 
 
+def create_luks_secret(vol_path, password):
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: The function supposed to be moved to avocado-vt?

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py [hunk 3]
```diff
@@ -2,12 +2,16 @@
 import logging
 import string
 import hashlib
+import base64
+import re
 
 from autotest.client.shared import utils
 from autotest.client.shared import error
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: please update autotest to avocado

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py [hunk 4]
```diff
@@ -54,6 +58,55 @@ def write_file(path):
...
+        raise exceptions.TestError("Fail to get newly created secret uuid")
+    logging.debug("Secret uuid %s", encryption_uuid)
+    # Set secret value.
+    secret_string = base64.b64encode(password)
+    ret = virsh.secret_set_value(encryption_uuid, secret_string)
+    utlv.check_exit_status(ret)
+    return encryption_uuid
+
+
+def create_luks_vol(pool_name, vol_name, sec_uuid, vol_arg):
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: could this be added in pre_vol?

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py (hunk 3 above)
Comment (lento-sun) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: I'll go through the whole tp-libvirt/libvirt/tests/src/virsh_cmd/volume later

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py (hunk 2 above)
Comment (lento-sun) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: Yes, I'll put it somewhere in avocado-vt

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py (hunk 4 above)
Comment (lento-sun) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py]: currently the scripts under this dir use different methods to get volume related arguments. And I create the create_luks_vol method with slight differences, that's definitely not a good way. I'll make modification later to make this method more common.

**Comment by lento-sun** [comment]
Now this depends on https://github.com/avocado-framework/avocado-vt/pull/1034

**Comment by lento-sun** [comment]
and now depends on https://github.com/avocado-framework/avocado-vt/pull/1036

**Comment by lento-sun** [comment]
Done, @dzhengfy @chunfuwen

**Pull request merged**